module/VuFind/src/VuFind/RateLimiter/RateLimiterManager.php

```diff
@@ -0,0 +1,299 @@
+            return $result;
+        } catch (\Exception $e) {
+            $this->logError((string)$e);
+        }
+        // Allow access on failure:
+        return $result;
+    }
+
+    /**
+     * Try to find a policy that matches an event
+     *
+     * @param MvcEvent $event Event
+     *
+     * @return ?string policy id or null if no match
+     */
+    protected function getPolicyIdForEvent(MvcEvent $event): ?string
+    {
+        foreach ($this->config['Policies'] ?? [] as $name => $settings) {
+            if (null !== ($loggedIn = $settings['loggedIn'] ?? null)) {
+                if ($loggedIn !== ($this->userId ? true : false)) {
+                    continue;
+                }
+            }
+            if ($ipRanges = $settings['ipRanges'] ?? null) {
+                if (!$this->ipUtils->isInRange($this->clientIp, (array)$ipRanges)) {
+                    continue;
+                }
+            }
+
+            if (!($filters = $settings['filters'] ?? null)) {
+                return $name;
+            }
+            foreach ($filters as $filter) {
+                if ($this->eventMatchesFilter($event, $filter)) {
+                    return $name;
+                }
+            }
+        }
+        return null;
+    }
+
+    /**
+     * Check if an event matches a filter
+     *
+     * @param MvcEvent $event  Event
+     * @param array    $filter Filter from configuration
+     *
+     * @return bool
+     */
+    protected function eventMatchesFilter(MvcEvent $event, array $filter): bool
+    {
+        $routeMatch = $event->getRouteMatch();
+        foreach ($filter as $param => $value) {
+            if ('name' === $param) {
+                if ($routeMatch?->getMatchedRouteName() !== $value) {
+                    return false;
+                }
+            } elseif (in_array($param, ['params', 'query', 'post'])) {
+                $req = $event->getRequest();
+                if ('params' === $param) {
+                    $allParams = $req->getPost()->toArray() + $req->getQuery()->toArray();
+                } elseif ('query' === $param) {
+                    $allParams = $req->getQuery()->toArray();
+                } else {
+                    $allParams = $req->getPost()->toArray();
+                }
+                foreach ($value as $key => $val) {
+                    if ($val !== $allParams[$key] ?? null) {
+                        return false;
+                    }
+                }
+            } elseif ($routeMatch?->getParam($param) !== $value) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Log a verbose debug message if configured
+     *
+     * @param string $msg Message
+     *
+     * @return void
+     */
+    protected function verboseDebug(string $msg): void
+    {
+        if (!($this->config['General']['verbose'] ?? false)) {
+            return;
```

Comment:
Would it be more concise to just put the log statement inside a positive if, instead of returning on a negative if?

Reply:
Done.